recipes/cpython/all/test_package/conanfile.py

```diff
@@ -94,18 +99,18 @@ def build(self):
         cmake.definitions["Python{}_FIND_STRATEGY".format(py_major)] = "LOCATION"
 
         if self.settings.compiler != "Visual Studio":
-            if tools.Version(self._py_version) < tools.Version("3.8"):
+            if Version(self._py_version) < Version("3.8"):
                 cmake.definitions["Python{}_FIND_ABI".format(py_major)] = self._cmake_abi.cmake_arg
 
-        with tools.environment_append(RunEnvironment(self).vars):
+        with legacy_tools.environment_append(RunEnvironment(self).vars):
             cmake.configure()
         cmake.build()
 
-        if not cross_building(self, skip_x64_x86=True):
+        if can_run(self):
             if self._supports_modules:
-                with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
-                    modsrcfolder = "py2" if tools.Version(self.deps_cpp_info["cpython"].version).major < "3" else "py3"
-                    tools.mkdir(os.path.join(self.build_folder, modsrcfolder))
+                with legacy_tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else legacy_tools.no_op():
```

Comment:
Why not use the `VCVars` generator?

Reply:
I'm afraid not, because it uses generate() method and should be combined to new generators. It should be updated, for sure, but generators should be updated together.